testsuite/MDAnalysisTests/topology/test_itp.py

```diff
@@ -156,27 +183,58 @@ class TestITPNoMass(ParserBase):
     def universe(self, filename):
         return mda.Universe(filename)
 
+    @pytest.fixture
+    def guessed_masses(self, top):
+        masses = np.array(top.masses.values)
+        names = np.array(top.names.values)
+
+        empty = []
+        for a in masses:
+            empty.append(np.isnan(a))
+
+        masses[empty] = DefaultGuesser(None).guess_masses(
+                DefaultGuesser(None).guess_types(names[empty]))
+        return masses
+
     def test_mass_guess(self, universe):
-        assert universe.atoms[0].mass not in ('', None)
+        assert not np.isnan(universe.atoms[0].mass)
 
 
 class TestITPAtomtypes(ParserBase):
     parser = mda.topology.ITPParser.ITPParser
     ref_filename = ITP_atomtypes
-    expected_attrs = ['ids', 'names', 'types', 'masses',
+    expected_attrs = ['ids', 'names', 'types',
                       'charges', 'chargegroups',
-                      'resids', 'resnames',
+                      'resids', 'resnames', 'masses'
                       'segids', 'moltypes', 'molnums',
                       'bonds', 'angles', 'dihedrals', 'impropers']
-    guessed_attrs = ['masses']
+    guessed_attrs = ['elements', ]
+
     expected_n_atoms = 4
     expected_n_residues = 1
     expected_n_segments = 1
 
+    @pytest.fixture
+    def guessed_typess(self, top):
```

Comment:
```suggestion
    def guessed_types(self, top):
```

Although see comments above about just using `top` in the `test_guessed_types` function

Reply:
since the `test_guessed_types` test has now been removed, is this still necessary?

Reply:
I just missed it.... I'll remove it